### PR TITLE
Close #341, Add ability to upload multiple files

### LINKED ISSUE #341: Step: upload a file
In preparation for https://github.com/SuffolkLITLab/docassemble-AssemblyLine/issues/139, create a step that will let a developer upload a file. My current thinking is that those files will be stored in the Sources folder as well. There may be puppeteer complications. Relevant resources:

- Puppeteer api: https://github.com/puppeteer/puppeteer/blob/v2.0.0/docs/api.md#elementhandleuploadfilefilepaths
- Strangely, differing answers on Stackoverflow: https://stackoverflow.com/questions/59273294/how-to-upload-file-with-js-puppeteer. There's an answer there that's independent of puppeteer if we get desperate enough.

### PR DESCRIPTION
**Pull request by plocket**
Should pass after https://github.com/plocket/docassemble-ALAutomatedTestingTests/pull/152 is reviewed as sufficient and merged.
Also suppress error from duplicate of [class="file-caption-name"]
input and increase `show if` timeout because test skipped setting
hidden field and all I can assume is that it didn't have enough
time to show up. Made #342, #343 for it.

Close #341

### CODE REVIEW

File: lib/scope.js [hunk 1]
```diff
@@ -991,13 +994,78 @@ module.exports = {
...
+        await scope.afterStep(scope, { waitForShowIf: true });
       }
     },  // Ends scope.setField.input()
   },
 
+  uploadFiles: async function ( scope, { handle, set_to }) {
+    /* Try to upload one or more files to a file-type input field. */
+
+    // Allow a typo where someone forgot to include a space after a comma?
+    let file_names = set_to.split(`,`);
```
Comment (niharikasingh) [inline lib/scope.js:1012]: In my head, this line belongs closer to the for loop below where we use the `file_names` variable for the first time.

File: lib/scope.js [hunk 2]
```diff
@@ -991,13 +994,78 @@ module.exports = {
...
+    }
+
+    let paths = [];
+    // If a file exist, add it to the list of files to be uploaded. If it doesn't, add a warning.
+    for ( let name of file_names ) {
+
+      // If the developer correctly used a ", ", there will be a space at the
+      // start of the string. Remove it. If they forgot the space, that's fine too.
+      name = name.replace(/^\s/, '');
+      let file_path = path.join( dir, name );
```
Comment (niharikasingh) [inline lib/scope.js:1048]: Looks like we're enforcing that all files listed in one line must be in the same directory (i.e., you can't have `some_png_1` in sources and `some_png_2` in features. We should document this.

**Review by niharikasingh — APPROVED** [review]

File: lib/scope.js (hunk 2 above)
Comment (plocket) [reply lib/scope.js:1048]: Great idea, thanks! #347

Comment (plocket) [reply lib/scope.js:1050]: Now see https://github.com/SuffolkLITLab/docassemble-AssemblyLine-documentation/issues/207

**Pull request merged**